src/utils/components/DiskModal/components/StorageProfileSettings/VolumeMode.tsx

```diff
@@ -1,58 +1,96 @@
 import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { ClaimPropertySets } from '@kubevirt-utils/types/storage';
 import { FormGroup, Radio } from '@patternfly/react-core';
 
 import { V1DiskFormState } from '../../utils/types';
-import {
-  ACCESS_MODE_FIELD,
-  STORAGE_CLASS_PROVIDER_FIELD,
-  VOLUME_MODE_FIELD,
-  VOLUMEMODE_FIELDID,
-} from '../utils/constants';
-import { getVolumeModeForProvisioner, VOLUME_MODE_RADIO_OPTIONS } from '../utils/modesMapping';
+import { ACCESS_MODE_FIELD, VOLUME_MODE_FIELD, VOLUMEMODE_FIELDID } from '../utils/constants';
+import { getAccessModesForVolume, VOLUME_MODE_RADIO_OPTIONS } from '../utils/modesMapping';
 
-const VolumeMode: FC = () => {
+import RecommendationLabel from './RecommendationLabel';
+
+import './ApplyStorageProfileSettings.scss';
+type VolumeModeProps = {
+  claimPropertySets: ClaimPropertySets;
+};
+
+const VolumeMode: FC<VolumeModeProps> = ({ claimPropertySets }) => {
   const { t } = useKubevirtTranslation();
 
-  const { control, setValue, watch } = useFormContext<V1DiskFormState>();
+  const { setValue, watch } = useFormContext<V1DiskFormState>();
 
   const volumeMode = watch(VOLUME_MODE_FIELD);
-  const accessModes = watch(ACCESS_MODE_FIELD) as V1beta1StorageSpecAccessModesEnum[];
-  const storageClassProvisioner = watch(STORAGE_CLASS_PROVIDER_FIELD);
+  const recommendedVolumeModes = uniq(
+    claimPropertySets
+      .map((it) => it.volumeMode)
+      .filter((mode) =>
+        Object.values(V1beta1StorageSpecVolumeModeEnum).some((key) => key === mode),
+      ),
+  );
 
-  const allowedVolumeModes = useMemo(
-    () => getVolumeModeForProvisioner(storageClassProvisioner, accessModes?.[0]),
-    [accessModes, storageClassProvisioner],
+  const setBothModes = useCallback(
+    (mode) => {
+      const accessModes = getAccessModesForVolume(claimPropertySets, mode);
+      setValue(VOLUME_MODE_FIELD, mode);
+      setValue(ACCESS_MODE_FIELD, accessModes.length > 0 ? [accessModes[0]] : undefined);
+    },
+    [claimPropertySets, setValue],
   );
 
   useEffect(() => {
-    if (!allowedVolumeModes?.includes(volumeMode as V1beta1StorageSpecVolumeModeEnum)) {
-      setValue(VOLUME_MODE_FIELD, allowedVolumeModes[0]);
+    if (volumeMode || !recommendedVolumeModes?.[0]) {
+      return;
     }
-  }, [allowedVolumeModes, volumeMode, setValue]);
+    const mostRecommended = recommendedVolumeModes[0];
+    setBothModes(mostRecommended);
+  }, [volumeMode, recommendedVolumeModes, setBothModes]);
 
   return (
-    <FormGroup fieldId={VOLUMEMODE_FIELDID} label={t('Volume Mode')}>
+    <FormGroup
+      labelHelp={
+        <HelpTextIcon
+          bodyContent={
+            <Trans ns="plugin__kubevirt-plugin" t={t}>
+              Learn more about
+              <Link target="_blank" to={documentationURL.STORAGE_PROFILES}>
+                {' '}
+                StorageProfile
+              </Link>
+              .
```

Comment:
?

Reply:
The message was suggested and accepted by the doc team - see https://issues.redhat.com/browse/CNV-55151?focusedId=26922579&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-26922579

Reply:
I have no strong opinion. The message was approved by the doc team so I think it's fine (and they do noticed missing dots in other places).